### DISCUSSION TITLE
[BUG]: unsupport Perl Syntax :'(?='

### DISCUSSION THREAD

**Discussion by KnightBlood** [alibaba/ilogtail#2129] [Help] [answered]
unsupport Perl Syntax :'(?='

`
processors:
  - Type: processor_parse
    SourceKey: content
    Regex:  (\d+-\d+-\d+ \d+:\d+:\d+.?)\|(\w+.?)\|([^\|]+)\|([^\|]+)\|([^\|]+)\|([^\|]+)\|([\s\S]*?)(?=\n\d+-\d+-\d+ \d+:\d+:\d+|\z)

`

**Comment by Takuka0311** [comment, marked as answer]
这是一个常见的问题，我会把它转到discussion。

Go的`processor_regex`插件使用的正则是"regexp"包，确实不支持Perl语言的所有正则表达式语法，包括' (?='。Go的"regexp"包支持的是RE2风格的正则表达式，复杂度过高的操作，比如回溯（如先行断言和后行断言）不被支持。

Go插件里，`processor_grok`使用的是"github.com/dlclark/regexp2"包，这个包是支持'(?=' 这样的操作的。此外，可以用C++的`processor_parse_regex_native `插件，也可以支持。

如果有兴趣贡献，可以把`processor_regex`使用的正则库改为"github.com/dlclark/regexp2"，不过这需要补充一定的ut和benchmark保证性能和稳定性不受影响
